src/v_video.c

```diff
@@ -849,9 +871,39 @@ void V_ScreenShot(char *format)
 #ifdef HAVE_LIBPNG
     if (png_screenshots)
     {
-    WritePNGfile(lbmname, I_VideoBuffer,
-                 SCREENWIDTH, SCREENHEIGHT,
-                 W_CacheLumpName (DEH_String("PLAYPAL"), PU_CACHE));
+        byte * buf, * oldbuf;
+        int oldpitch;
+
+        // Prepare a temporary buffer to write the screenshot to
+        buf = malloc(screenshot_width * screenshot_height);
+
+        if(!buf)
+        {
+            return;
+        }
+
+        // Ideally we'd do this only once, but V_Init is called too early
+        // and screenshot_width/height haven't been read from the config
+        V_InitScreenShotScaler();
+
+        // back up the scaling system's current output buffer settings
+        oldbuf = I_GetScaleDestBuf();
```

Comment:
Finally got round to looking at this again. Here are my thoughts:

How about we add a wrapper function around ```->DrawScreen```? Callers shouldn't invoke it directly but instead call a function with ```(uint8_t *buf, unsigned int pitch)``` as arguments? And the wrapper can then set the globals appropriately before calling the scale function?

I also wonder if part of ```I_FindScreenMode()``` should be moved into i_scale.c too. Not sure though.